diff --git a/dist/src/index.js b/dist/src/index.js
index 4e20dd93082d8459e65eb740aff7c6da9f5d063b..46b34753df39ddc7171565a07b1b02ae16a714ba 100644
--- a/dist/src/index.js
+++ b/dist/src/index.js
@@ -10,7 +10,13 @@
 // @ts-ignore untyped from Emscripten
 import loadYoga from '../binaries/yoga-wasm-base64-esm.js';
 import wrapAssembly from "./wrapAssembly.js";
-const Yoga = wrapAssembly(await loadYoga());
+
+// Initialize Yoga without top-level await
+let Yoga;
+(async () => {
+    Yoga = wrapAssembly(await loadYoga());
+})();
+
 export default Yoga;
 export * from "./generated/YGEnums.js";
 //# sourceMappingURL=index.js.map
\ No newline at end of file
